pkgs/development/tools/documentation/adrdox/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchFromGitHub, dmd }:
+
+stdenv.mkDerivation rec {
+  pname = "adrdox";
+  version = "v1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "adamdruppe";
+    repo = "adrdox";
+    rev = version;
+    sha256 = "1w0y158ab36385jhg6q9xjlb8cilfwimlh43q86n40l10vhn4bk9";
+  };
+
+  buildInputs = [ dmd ];
+
+  # We need a custom install phase because the makefile lacks an install
+  # target. If a future version of adrdox gets an install target, the custom
+  # install phase may be removed.
+  installPhase = ''
```

Comment:
I think it is a good idea to put `runHook preInstall` at the start and `runHook postInstall` at the end so that people can override these stages if they want to.

Reply:
Added.